### PR TITLE
Increase testcoverage and refactor #3

### PR DESCRIPTION
**Pull request by bastilian**
* Update react-router-com to support hooks
 * Add tests for:
   * CompliancePolicies
   * DownloadTableButton
   * BusinessObjectiveField
   * EditPolicy
   * ProfileThresholdField
   * SystemDetails
   * InventoryDetails
* Rewrites the SystemsDetails component to use hooks
* + minor fixes and updates

This brings us to about 66% coverage, removing the ComplianceRemidiationsButton and replacing it with the frontend-component version will bring us to about 75% code coverage.

The remaining 25% of uncovered code are mostly minor functions, but a branch with tests covering those is in progress.

### CODE REVIEW

File: src/SmartComponents/CompliancePolicies/CompliancePolicies.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ import {
     GridItem
 } from '@patternfly/react-core';
 
-const QUERY = gql`
+export const QUERY = gql`
```
[the remark this answers is not in the document]
Comment (bastilian) [reply src/SmartComponents/CompliancePolicies/CompliancePolicies.js]: This export actually not needed, I'll update that. But yeah, if we'd export them a more decriptive name would be good.

File: src/SmartComponents/ComplianceRemediationButton/ComplianceRemediationButton.js [hunk 2]
```diff
@@ -1,3 +1,6 @@
+/*
+    TODO: Replace with frontend-components version
```
[the remark this answers is not in the document]
Comment (bastilian) [reply src/SmartComponents/ComplianceRemediationButton/ComplianceRemediationButton.js]: Not yet, I'll create one.

File: src/SmartComponents/DownloadTableButton/DownloadTableButton.js [hunk 3]
```diff
@@ -5,7 +5,7 @@ import { connect } from 'react-redux';
 import { Dropdown, KebabToggle, DropdownItem, Tooltip } from '@patternfly/react-core';
 import { exportToCSV } from '../../store/ActionTypes.js';
 
-class DownloadTableButton extends React.Component {
+export class DownloadTableButton extends React.Component {
```
[the remark this answers is not in the document]
Comment (bastilian) [reply src/SmartComponents/DownloadTableButton/DownloadTableButton.js]: We do a `export default` of the DownloadTableButton, however, that default export is connected with redux. That is why we export the "raw" component to make testing it easier.

**Comment by bastilian** [comment]
@akofink the export is updated. I also tested it with the latest chrome and I don't see the issue with `browser.js`, it does show the issue with the flickering table that is also present in CI/QA envs.

File: src/SmartComponents/ComplianceRemediationButton/ComplianceRemediationButton.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (bastilian) [reply src/SmartComponents/ComplianceRemediationButton/ComplianceRemediationButton.js]: ✅  Issue created

File: src/SmartComponents/DownloadTableButton/DownloadTableButton.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (akofink) [reply src/SmartComponents/DownloadTableButton/DownloadTableButton.js]: Ah, gotcha :+1:

**Pull request merged**